### ISSUE TITLE
Flaky Project Model Server Test

### ISSUE THREAD

**Issue by piotrpMSFT** [dotnet/cli#2996]
## Steps to reproduce
http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0/job/debug_debian8.2_x64_prtest/664/consoleFull

## Expected  behavior
Test passes or fails reliably

## Actual behavior
Test occasionally fails with 
A unexpected exception occurred: 
````
/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0/debug_debian8.2_x64_prtest/test/dotnet-projectmodel-server.Tests/bin/Debug/netcoreapp1.0/RecoverFromGlobalError/EmptyConsoleApp/project.json(12,1): Error: Microsoft.DotNet.ProjectModel.FileFormatException: Additional text encountered after finished reading JSON content: }. Path '', line 12, position 1. ---> Newtonsoft.Json.JsonReaderException: Additional text encountered after finished reading JSON content: }. Path '', line 12, position 1.
   at Newtonsoft.Json.JsonTextReader.ReadInternal()
   at Newtonsoft.Json.JsonTextReader.Read()
   at Microsoft.DotNet.ProjectModel.ProjectReader.ReadProject(Stream stream, String projectName, String projectPath, ProjectReaderSettings settings)
   at Microsoft.DotNet.ProjectModel.ProjectReader.TryGetProject(String path, Project& project, ProjectReaderSettings settings)
   --- End of inner exception stack trace ---
   at Microsoft.DotNet.ProjectModel.ProjectReader.TryGetProject(String path, Project& project, ProjectReaderSettings settings)
   at Microsoft.DotNet.ProjectModel.Workspace.AddProjectEntry(String projectDirectory, FileModelEntry`1 currentEntry)
   at Microsoft.DotNet.ProjectModel.Workspace.<GetProjectCore>b__11_0(String key)
   at System.Collections.Concurrent.ConcurrentDictionary`2.AddOrUpdate(TKey key, Func`2 addValueFactory, Func`3 updateValueFactory)
   at Microsoft.DotNet.ProjectModel.Workspace.GetProjectCore(String projectDirectory)
   at Microsoft.DotNet.ProjectModel.Workspace.AddProjectContextEntry(String projectDirectory, ProjectContextCollection currentEntry)
   at Microsoft.DotNet.ProjectModel.Workspace.<GetProjectContextCollection>b__8_0(String key)
   at System.Collections.Concurrent.ConcurrentDictionary`2.AddOrUpdate(TKey key, Func`2 addValueFactory, Func`3 updateValueFactory)
   at Microsoft.DotNet.ProjectModel.Workspace.GetProjectContextCollection(String projectPath)
   at Microsoft.DotNet.ProjectModel.Server.ProjectSnapshot.Create(String projectDirectory, String configuration, DesignTimeWorkspace workspaceContext, IReadOnlyList`1 projectSearchPaths)
   at Microsoft.DotNet.ProjectModel.Server.ProjectManager.UpdateProject()
   at Microsoft.DotNet.ProjectModel.Server.ProjectManager.DoProcessLoop()
   at Microsoft.DotNet.ProjectModel.Server.ProjectManager.ProcessLoop()
````

**Comment by troydai** [comment]
@piotrpMSFT 

The error message is correct. This is a negative case. It adds unused characters to project.json to trigger a global error. The test doesn't seem to fail according to the test report of the build you pointed at: http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0/job/debug_debian8.2_x64_prtest/664/testReport/

**Comment by piotrpMSFT** [comment]
Ok, I suppose it's not that specific test. The output of that set of tests is quite verbose with lots of intended failures. Let's start here, then:

=== TEST EXECUTION SUMMARY ===
   dotnet-projectmodel-server.Tests  Total: 23, Errors: 0, Failed: 1, Skipped: 1, Time: 149.893s

**Comment by troydai** [comment]
So we're actually talking about this one: 
```
MESSAGE:
Dependencies message doesn't exists. Expected: False Actual: True
+++++++++++++++++++
STACK TRACE:
at Microsoft.DotNet.ProjectModel.Server.Tests.DthMessageCollectionExtension.RetrieveSingleMessage(IEnumerable`1 messages, String typename) at Microsoft.DotNet.ProjectModel.Server.Tests.DthTests.RemoveMSBuildDependencyFromProjectJson()
```

Does it become flaky recently or it is always like this?
Do we run tests in parallel?

**Comment by piotrpMSFT** [comment]
it fails occasionally. We do run tests in parallel in some contexts.

**Comment by troydai** [comment]
How can I tell if project model server tests are run in parallel?

**Comment by piotrpMSFT** [comment]
Fixed by #3142
